src/app/common/components/FilterToolbar/FilterToolbar.tsx

```diff
@@ -24,18 +25,18 @@ export interface OptionPropsWithKey extends SelectOptionProps {
   key: string;
 }
 
-export interface IBasicFilterCategory {
+export interface IBasicFilterCategory<T> {
   key: string;
   title: string;
   type: FilterType; // If we want to support arbitrary filter types, this could be a React node that consumes context instead of an enum
-  getItemValue?: (item: any) => any;
+  getItemValue?: (item: T) => string | boolean;
 }
 
-export interface ISelectFilterCategory extends IBasicFilterCategory {
+export interface ISelectFilterCategory extends IBasicFilterCategory<IPlan> {
```

Comment:
This should be `ISelectFilterCategory<T> extends IBasicFilterCategory<T>`, the `IPlan` can be passed in when it's used.

Reply:
I needs a real type here because it's passing as parameter to the `IBasicFilterCategory` interface

Reply:
Right, ok. The FilterToolbar file shouldn't be specific to Plans though, and `ISelectFilterCategory<T> extends IBasicFilterCategory<T>` will define this interface as a generic one. You can then use `ISelectFilterCategory<IPlan>` in the actual table component and that will define what `T` means here, so it will make its way to the `IBasicFilterCategory`.

Reply:
To get this to work you will actually also need to add generics to `FilterCategory`, turn it into `export type FilterCategory<T> = ISearchFilterCategory<T> | ISelectFilterCategory<T>;` and then use `FilterCategory<IPlan>[]` as the type of the categories array in the table. That should get you the right types passed down.

Reply:
Works fine!